src/components/v5/common/ActionSidebar/partials/TokenSelect/partials/TokenSearchSelect/TokenSearchSelect.tsx

```diff
@@ -0,0 +1,152 @@
+import { Binoculars } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import debounce from 'lodash/debounce';
+import React, {
+  type FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { useMobile } from '~hooks/index.ts';
+import { SpinnerLoader } from '~shared/Preloaders/index.ts';
+import { formatText } from '~utils/intl.ts';
+import EmptyContent from '~v5/common/EmptyContent/index.ts';
+import MenuContainer from '~v5/shared/MenuContainer/MenuContainer.tsx';
+import Portal from '~v5/shared/Portal/index.ts';
+import SearchInput from '~v5/shared/SearchSelect/partials/SearchInput/SearchInput.tsx';
+
+import TokenSearchItem from '../TokenSearchItem/TokenSearchItem.tsx';
+
+import { useSearchSelect } from './hooks.ts';
+import { type TokenSearchSelectProps } from './types.ts';
+
+const displayName =
+  'v5.common.ActionsContent.partials.TokenSelect.partials.TokenSearchSelect';
+
+const TokenSearchSelect = React.forwardRef<
+  HTMLDivElement,
+  TokenSearchSelectProps
+>(
+  (
+    {
+      items,
+      onSelect,
+      isLoading,
+      hideSearchOnMobile,
+      onSearch,
+      showEmptyContent = true,
+      state,
+      message,
+      additionalButtons,
+      className,
+    },
+    ref,
+  ) => {
+    const [searchValue, setSearchValue] = useState('');
+    const [debouncedSearchValue, setDebouncedSearchValue] = useState('');
+    const isMobile = useMobile();
+    const filteredList = useSearchSelect(items, debouncedSearchValue);
+
+    useEffect(() => {
+      const debounceTimeout = setTimeout(() => {
+        setDebouncedSearchValue(searchValue);
+      }, 500);
```

Comment:
Not sure why we need to set up another 500 millisecond timeout here when we can just fully rely on the `debounce ()` function from `lodash` to handle it for us, or maybe there some edge-case I'm not aware of 🤔 But was thinking we can have a more concise and leaner approach to this:

```js
const [searchValue, setSearchValue] = useState('');
const [debouncedSearchValue, setDebouncedSearchValue] = useState('');
const isMobile = useMobile();
const filteredList = useSearchSelect(items, debouncedSearchValue);

// The only handler for delaying a state update for `debouncedSearchValue`
const debouncedSearchValueSetter = useCallback(
  debounce((value) => setDebouncedSearchValue(value), 500),
  [],
);

useEffect(() => {
  debouncedSetSearchValue(searchValue);

  return () => {
    // Maybe this was the behaviour you intended instead of a setTimeout?
    debouncedSetSearchValue.cancel();
  };
}, [searchValue, debouncedSetSearchValue]);

const onChange = useCallback(
  (value) => {
    onSearch?.(value);
    setSearchValue(value);
    // Removed the if statement here because in my example, we don't need `handleSearch` anymore
    // since the debounce logic is now handled by the useEffect when a state update is performed for `searchValue`
    // The sequence is:
    // 1. `setSearchValue(value)` <-- state update is performed for `searchValue`
    // 2. `useEffect` picks this up and invokes `debouncedSetSearchValue(searchValue)`
    // 3. debounce is performed within `debouncedSetSearchValue()`
  },
  [onSearch],
);
```

In my example, I've removed the following which to me felt redundant:
- `useMemo`
- `setTimeout`